metaflow/plugins/argo/argo_events_decorator.py

```diff
@@ -0,0 +1,154 @@
+                        raise MetaflowException(
+                            "One or more events in *events* attribute in *@trigger* "
+                            "decorator have an incorrect format. Supported format "
+                            "is dictionary - \n"
+                            "@trigger(events=[{'name': 'foo', 'parameters': {'alpha': "
+                            "'beta'}}, {'name': 'bar', 'parameters': "
+                            "{'gamma': 'kappa'}}])"
+                        )
+            else:
+                raise MetaflowException(
+                    "Incorrect format for *events* attribute in *@trigger* decorator. "
+                    "Supported format is list - \n"
+                    "@trigger(events=[{'name': 'foo', 'parameters': {'alpha': "
+                    "'beta'}}, {'name': 'bar', 'parameters': "
+                    "{'gamma': 'kappa'}}])"
+                )
+
+        if not self.triggers:
+            raise MetaflowException("No event(s) specified in *@trigger* decorator.")
+
+        # same event shouldn't occur more than once
+        names = [x["name"] for x in self.triggers]
+        if len(names) != len(set(names)):
+            raise MetaflowException(
+                "Duplicate event names defined in *@trigger* decorator."
+            )
+
+
+class TriggerOnFinishDecorator(FlowDecorator):
+    name = "trigger_on_finish"
+    defaults = {
+        "flow": None,
+        "branch": None,
+        "project": None,
+        "flows": [],
+        "options": {},
+    }
+
+    def flow_init(
+        self, flow, graph, environment, flow_datastore, metadata, logger, echo, options
+    ):
+        self.triggers = []
+        if sum(map(bool, (self.attributes["flow"], self.attributes["flows"]))) > 1:
+            raise MetaflowException(
+                "Specify only one of *flow* or *flows* "
+                "attributes in *@trigger_on_finish* decorator."
+            )
+        elif self.attributes["flow"]:
+            # flow supports the format @trigger_on_finish(flow='FooFlow')
+            if is_stringish(self.attributes["flow"]):
+                self.triggers.append(self.attributes)
+            else:
+                raise MetaflowException(
+                    "Incorrect format for *flow* attribute in *@trigger_on_finish* "
+                    " decorator. Supported format is string - \n"
+                    "@trigger_on_finish(flow='FooFlow')"
+                )
+        elif self.attributes["flows"]:
+            # flows attribute supports the following formats -
+            #     1. flows=['FooFlow', 'BarFlow']
+            if isinstance(self.attributes["flows"], list):
+                for flow in self.attributes["flows"]:
+                    if is_stringish(flow):
+                        self.triggers.append(
+                            {
+                                "flow": flow,
+                                "project": self.attributes["project"],
```

Comment:
This seems incorrect -- it should use the default project/branch. It makes sense to make this change here as that is the behavior across implementations.

Reply:
we don't have access to the values from current here. `triggers` only records what the user specifies and the default value handling is taken care of by the implementation where we have access to current.